### ISSUE TITLE
"Artifacts are from different compiler runs" incorrect error using Windows Git Bash

### ISSUE THREAD

**Issue by andyunslashed** [OpenZeppelin/openzeppelin-upgrades#217]
= Problem =

I have been trying to use deployProxy on WIndows (10, using Git Bash) but keeping getting the above error; my colleagues using Linux and MacOS have had no such issues, using the same versions of all tools as me.

= Investigation =
We investigated the code that produces the error (reconstructSolcInputOutput in packages/plugin-truffle/src/validate.ts) and have identified the problem.

importDir.absolutePath is of the format '@openzeppelin/contracts-ethereum-package/contracts/Initializable.sol' but the corresponding key in sourceUnitId is '@openzeppelin\\contracts-ethereum-package\\contracts\\Initializable.sol' so importedUnitId is undefined and the sourceUnit will never match.

Similarly, when dealing with local files, "C://" does not match with "\C\".

In short, the error string is incorrect in this case and misleading to users about what the actual issue is.

= Workaround =

Change
```
const importedUnitId = sourceUnitId[importDir.absolutePath];
```
to
```
let importedUnitId = sourceUnitId[importDir.absolutePath];
if (undefined === importedUnitId) {
  const path = importDir.absolutePath.replace(/\/C\//g, 'C:/').replace(/\//g, '\\');
  importedUnitId = sourceUnitId[path];
}
```

**Comment by frangio** [comment]
Thank you so much @andyunslashed! We've had this issue reported on Windows before but had no idea about the cause.

I will try to publish a fix as soon as possible.

**Comment by andyunslashed** [comment]
No problem! :)

Looking forward to seeing this fixed properly!

**Comment by frangio** [comment]
@andyunslashed I ended up fixing this in a different way, but it should work. Would you be able to test the prerelease version `@openzeppelin/truffle-upgrades@1.2.3-0` and let me know if it works?

**Comment by andyunslashed** [comment]
It works!

On behalf of all truffle contract upgrading Windows users everywhere (probably only 3 of us!) - thank you!
